src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -16,6 +19,18 @@
      */
     Optional<Book> findByAuthor(String author);
 
+    /**
+     * Method that obtain all books by your genre, publisher and year
+     *
+     * @param genre:     Genre of the book (String)
+     * @param publisher: Publisher of the book (String)
+     * @param year:      Year of the book (String)
+     * @return {@link Book} list
+     */
+    @Query("SELECT b FROM Book b WHERE b.genre = :genre and b.publisher = :publisher and b.year = :year")
```

Comment:
If the genre value is NULL, the filter is not applied?

Reply:
This query will run if it has all the parameters.

Reply:
That's right! However, the card asks you to provide a method where you can have optional parameters (when they are NULL) and in that case that filter should not be applied.